pkg/generate/app/imagelookup.go

```diff
@@ -154,7 +154,7 @@ func (r DockerRegistryResolver) Resolve(value string) (*ComponentMatch, error) {
 
 	from := ref.Registry
 	if len(ref.Registry) == 0 {
-		ref.Registry = "Docker Hub"
+		from = "Docker Hub"
```

Comment:
Yes, for a pure docker image that gets passed in on the cli, the default behavior for it being on the hub or in a private registry is to create an image stream with a docker image repository set.  If the ref is an existing image stream, you should point directly to the image stream from the DC or BC via From: "ImageStreamTag", Namespace:, Name:

----- Original Message -----
> > @@ -154,7 +154,7 @@ func (r DockerRegistryResolver) Resolve(value string)
> > (*ComponentMatch, error) {
> >  
> >  	from := ref.Registry
> >  	if len(ref.Registry) == 0 {
> > -		ref.Registry = "Docker Hub"
> > +		from = "Docker Hub"
> 
> Besides not wanting to hard code anything specific to the Docker Hub in here,
> this isn't going to work because `ref.Registry` might be populated. If it is
> populated, it could point to the Hub (docker.io), it could point to an
> external registry (myregistry.com), or it could point to our integrated
> registry (172.30.x.x:5000 or a route for it). I think you should be creating
> the input ImageStream if the ref has no registry or if its registry doesn't
> point to the OpenShift integrated registry. @smarterclayton thoughts?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2089/files#r29946356